lib/spack/spack/binary_distribution.py

```diff
@@ -67,8 +65,8 @@ class BinaryCacheIndex(object):
     mean we should have paid the price to update the cache earlier?
     """
 
-    def __init__(self, cache_root=None):
-        self._cache_root = cache_root or default_binary_index_root
+    def __init__(self, cache_root):
+        self._cache_root = cache_root
         self._index_cache_root = os.path.join(self._cache_root, 'indices')
```

Comment:
this nests things `~/.spack/cache/indices/indices` -- needs one fewer `indices`

Reply:
Oh! Thanks.

Reply:
I have resolved this by removing `self._cache_root` entirely and setting `self._index_cache_root = cache_root` directly.